src/hdmf/build/classgenerator.py

```diff
@@ -35,15 +36,17 @@ def register_generator(self, **kwargs):
             {'name': 'spec', 'type': BaseStorageSpec, 'doc': ''},
             {'name': 'parent_cls', 'type': type, 'doc': ''},
             {'name': 'attr_names', 'type': dict, 'doc': ''},
+            {'name': 'post_init_method', 'type': types.FunctionType, 'default': None,
```

Comment:
```suggestion
            {'name': 'post_init_method', 'type': types.Callable, 'default': None,
```

`typing.Callable` is generally recommended over `types.FunctionType` because it is more general and can be parameterized with the input argument types and return type

Reply:
Is there a callable case that's not a function that we would want included by using callable type?

Reply:
In rare cases, a user may want to make a class callable by implementing `__call__`. There are some examples here: https://realpython.com/python-callable-instances/#putting-pythons-__call__-into-action . No reason to exclude that, in my opinion.

Reply:
I tried this, but it says Callable is not recognized from the types package. Are you sure that is the proper syntax?

Reply:
Sorry. It's in the `typing` package